app/src/main/java/org/fossasia/openevent/adapters/LocationsListAdapter.java

```diff
@@ -105,18 +107,27 @@ public void moveItem(int fromPosition, int toPosition) {
         notifyItemMoved(fromPosition, toPosition);
     }
 
-    class Viewholder extends RecyclerView.ViewHolder {
+    public static interface AdapterOnClickHandler {
+        void onClick(int pos, Viewholder vh);
+    }
+
+    public static class Viewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
         TextView name;
         TextView floor;
 
         public Viewholder(View itemView) {
             super(itemView);
             itemView.setClickable(true);
-
+            itemView.setOnClickListener(this);
             name = (TextView) itemView.findViewById(R.id.location_name);
             floor = (TextView) itemView.findViewById(R.id.location_floor);
 
         }
 
+        @Override
+        public void onClick(View view) {
+            clickHandler.onClick(getAdapterPosition(), this);
```

Comment:
Can I Implement in this way? I haven't used any explicit reference from outer class in inner class 
```
@Override
public void onBindViewHolder(Viewholder holder, int position) {
    Session current = sessions.get(position);
    String title = current.getTitle();
    String summary = current.getSummary();
    holder.sessionName.setText(title);
    holder.sessionSummary.setText(summary);
    holder.itemView.setOnClickListener(new myOnClickListener(position,clickHandler));
}
```

```
private static class myOnClickListener implements View.OnClickListener {
        private int position;
        private AdapterOnClickHandler clickHandler;
        myOnClickListener(int pos,AdapterOnClickHandler ch){
            position=pos;
            clickHandler=ch;
        }
        @Override
        public void onClick(View v) {
            clickHandler.onClick(position);
        }
    }
```

```
public static class Viewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView sessionName;
        TextView sessionSummary;

        public Viewholder(View itemView) {
            super(itemView);
            itemView.setClickable(true);
            itemView.setOnClickListener(this);
            sessionName = (TextView) itemView.findViewById(R.id.session_title);
            sessionSummary = (TextView) itemView.findViewById(R.id.session_abstract);

        }

        @Override
        public void onClick(View view) {

        }
    }
```